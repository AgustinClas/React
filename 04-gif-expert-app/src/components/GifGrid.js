import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';

export const GifGrid = ({category}) => {

    const {loading, data:images} = useFetchGifs(category);

    return (
        <div className = "GifGrid">
            <h3>{category}</h3>
            {loading && <p>Cargando</p>} 
            <div className='card-grid'>
                {
                    images.map((img) =>(
                        <GifGridItem 
                        {...img}
                        key = {img.id}
                        />
                        ))
                }
            </div> 
        </div>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}