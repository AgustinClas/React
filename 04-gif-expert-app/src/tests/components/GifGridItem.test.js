import React from "react";
import {shallow} from 'enzyme';
import '@testing-library/jest-dom';
import { GifGridItem } from "../../components/GifGridItem";

describe('GifGridItem', () => {

    const title = "titulo";
    const url = "https://localhost/foto.jpg"

    const wrapper = shallow(<GifGridItem title = {title} url = {url}/>);

    test('debe mostrar <GifGridItem/> correctamente', () => {
        

        expect( wrapper ).toMatchSnapshot();

    })

    test('debe mostrar un parrafo con el titulo', () => {

        const p = wrapper.find('p');

        expect( p.text().trim() ).toBe( title );

    })

    test('debe coincidir los valores cargados en la imagen', () => {

        const img = wrapper.find('img');

        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);


    })

    test('debe tener la clase correspondiente', () => {

        const div = wrapper.find('div');
        const className = div.prop("className");
        expect(className.includes('animate__fadeInDown')).toBe(true);
    });
    
});
