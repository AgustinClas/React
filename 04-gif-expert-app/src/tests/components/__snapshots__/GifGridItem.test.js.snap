// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGridItem debe mostrar <GifGridItem/> correctamente 1`] = `
<div
  className="card animate__animated animate__fadeInDown"
>
  <p
    className="TituloGifs"
  >
    titulo
  </p>
  <img
    alt="titulo"
    src="https://localhost/foto.jpg"
  />
</div>
`;
